astropy/cosmology/_src/utils.py

```diff
@@ -92,7 +97,9 @@ def all_cls_vars(obj: object | type, /) -> dict[str, Any]:
     return functools.reduce(operator.__or__, map(vars, cls.mro()[::-1]))
 
 
-def deprecated_keywords(*kws, since):
+def deprecated_keywords(
+    *kws: str, since: str | float | tuple[str | float, ...]
+) -> Callable[[_F], _F]:
```

Comment:
Why is this not using [`ParamSpec`](https://docs.python.org/3.11/library/typing.html#typing.ParamSpec)?